python/pyspark/ml/_typing.pyi

```diff
@@ -75,3 +78,5 @@ RankingEvaluatorMetricType = Union[
     Literal["ndcgAtK"],
     Literal["recallAtK"],
 ]
+
+VectorLike = Union[ndarray, Vector, List[float], Tuple[float, ...]]
```

Comment:
I am not super happy about adding this here, but unless we plant to revisit something like #14263, it is probably the way to go.

I was thinking about `np.typing.ArrayLike`, but we'd still need unions with our vectors, and it has much bigger impact on the hints, than simply changing the the alias.